File: src/main/java/destiny/core/chinese/ziwei/FlowHourBranchImpl.java
/**
 * Created by smallufo on 2017-04-18.
 */
package destiny.core.chinese.ziwei;

import destiny.core.chinese.Branch;

import java.io.Serializable;


/**
 * 計算流時命宮
 *
 * 設定於 {@link ZSettings.FlowHour#FLOW_HOUR_FIXED}
 * 固定以該時辰當作命宮
 */
public class FlowHourBranchImpl implements IFlowHour , Serializable {

  @Override
  public Branch getFlowHour(Branch hour, Branch flowDayMainHour) {
    return hour;
  }
}

File: src/main/java/destiny/core/chinese/ziwei/FlowHourBranchImpl.kt
/**
 * Created by smallufo on 2017-04-18.
 */
package destiny.core.chinese.ziwei

import destiny.core.Descriptive
import destiny.core.chinese.Branch
import destiny.tools.asDescriptive
import java.io.Serializable


/**
 * 計算流時命宮
 *
 * 固定以該時辰當作命宮
 */
class FlowHourBranchImpl : IFlowHour,
                           Descriptive by FlowHour.Branch.asDescriptive(),
                           Serializable {

  override fun getFlowHour(hour: Branch, flowDayMainHour: Branch): Branch {
    return hour
  }

}
